### PR TITLE
feat: added prometheus rw2 translation for gauges

### PR DESCRIPTION
**Pull request by jmichalek132**
#### Description

Follow up from #35703.

Draft starting the work on adding support for remote write 2.0 in the translation package.
Adding support for translating gauges.

This is first iteration and to keep the PR small
* we don't handle duplicate metrics
* only support gauges
* don't handle other labels than metric name
* don't handle exemplars
*  don't handle metadata

#### Link to tracking issue #33661
Fixes

### CODE REVIEW

File: .chloggen/jm-prom-translation-rw2-gauges-support.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/translator/prometheusremotewrite
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: add FromMetricsV2
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [33661]
```
Comment (ArthurSens) [inline .chloggen/jm-prom-translation-rw2-gauges-support.yaml:13]: It should be 35734 now

File: pkg/translator/prometheusremotewrite/number_data_points_v2_test.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+			labels.Hash(): {
+				LabelsRefs: []uint32{1, 2},
+				Samples: []writev2.Sample{
+					{Timestamp: convertTimeStamp(pcommon.Timestamp(ts)), Value: 2},
+				},
+			},
+		}
+	}
+
+	t.Run("gauge duplicate", func(t *testing.T) {
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/number_data_points_v2_test.go:153]: t.Run usually is used when we're testing multiple cases in a loop.

There is only one case here so there's no need for t.Run :)

File: pkg/translator/prometheusremotewrite/number_data_points_v2_test.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+			diff := cmp.Diff(w, converter.unique, cmpopts.EquateNaNs())
+			assert.Empty(t, diff)
+
+		})
+	}
+}
+
+// Right now we are not handling duplicates, the second one will just overwrite the first one as this test case shows
+// TODO handle duplicates
+func TestPrometheusConverterV2_addGaugeNumberDataPointsDuplicate(t *testing.T) {
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/number_data_points_v2_test.go:124]: Really nice that you have a test proving that we have an undesired behavior!

May I suggest a change in wording?

```suggestion
// Right now we are not handling duplicates, the second one will just overwrite the first one as this test case shows
// In follow-up PRs we plan to start handling conflicts and this test will be updated to reflect the new behavior.
func TestPrometheusConverterV2_addGaugeNumberDataPointsDuplicate(t *testing.T) {
```

File: pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	wanted := want()
+	require.NoError(t, err)
+	require.NotNil(t, tsMap)
+	require.Equal(t, wanted, tsMap)
+	require.NotNil(t, symbolsTable)
+
+}
+
+func createExportRequestWithTimestamp(resourceAttributeCount int, histogramCount int, nonHistogramCount int, labelsPerMetric int, exemplarsPerSeries int, timestamp pcommon.Timestamp) pmetricotlp.ExportRequest {
+	request := pmetricotlp.NewExportRequest()
```
Comment (ArthurSens) [inline pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go:50]: I'm a bit confused with the motivation for this method 🤔 

We're only implementing Gauges, so why do we have something that generates histograms?

**Review by ArthurSens — COMMENTED** [review]
Nice work!

The PR is basically ready for merge to be honest, I just have a few questions and comments around tests

File: pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go (hunk 4 above)
Comment (jmichalek132) [reply pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go:50]: it's a copy of existing method, I just added support for passing the timestamp, I could potentially just update the existing one.

File: .chloggen/jm-prom-translation-rw2-gauges-support.yaml (hunk 1 above)
Comment (jmichalek132) [reply .chloggen/jm-prom-translation-rw2-gauges-support.yaml:13]: it should be the number of a github issues if one exists which it this case it does.

File: pkg/translator/prometheusremotewrite/number_data_points_v2_test.go (hunk 2 above)
Comment (jmichalek132) [reply pkg/translator/prometheusremotewrite/number_data_points_v2_test.go]: fixed

File: pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go (hunk 4 above)
Comment (dashpole) [reply pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go:50]: +1 to updating the existing method instead of duplicating it

**Review by dashpole — APPROVED** [review]
lgtm after outstanding comment is resolved

File: pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go (hunk 4 above)
Comment (jmichalek132) [reply pkg/translator/prometheusremotewrite/metrics_to_prw_v2_test.go]: done

**Review by dashpole — APPROVED** [review]

**Review by ArthurSens — APPROVED** [review]
:shipit:

**Pull request merged**